src/PKSim.Presentation/Services/RenameBuildingBlockTask.cs

```diff
@@ -205,14 +205,24 @@ private void renameFormulaPathReferencingContainerInSimulation(Simulation simula
          renamedContainer.GetAllChildren<IFormulaUsable>().Each(reference =>
          {
             var allObjectReferencing = _objectReferencingRetriever.AllUsingFormulaReferencing(reference, simulation.Model);
-            renameFormulaPathInFormulas(allObjectReferencing, renamedContainer.Name, oldContainerName);
+            renameFormulaPathInFormulas(allObjectReferencing, reference, renamedContainer.Name, oldContainerName);
          });
       }
 
-      private void renameFormulaPathInFormulas(IEnumerable<IUsingFormula> referencingFormulas, string newName, string oldName)
+      private void renameFormulaPathInFormulas(IEnumerable<IUsingFormula> referencingFormulas, IFormulaUsable reference, string newName, string oldName)
       {
-         referencingFormulas.Select(x => x.Formula).SelectMany(x => x.ObjectPaths)
-            .Each(objecPath => objecPath.Replace(oldName, newName));
+         foreach (var formula in referencingFormulas.Select(x => x.Formula).Where(x => x.AreReferencesResolved))
+         {
+            //Reference is always found by construction
+            var referenceIndex = formula.ObjectReferences.Select(x => x.Object).IndexOf(reference);
+            var objectPath = formula.ObjectPaths[referenceIndex];
+
+            //The reference is used. The path needs to be updated only if it is a path referencing the old name
+            if(!objectPath.Contains(oldName))
+               continue;
+
+            objectPath.ReplaceWith(reference.EntityPath().ToPathArray());
```

Comment:
Here is the one change. I am swapping the path directly with a new one instead of messing around with rename